src/ml/neural_net/mps_layer_conv_padding.m

```diff
@@ -0,0 +1,57 @@
+#include <ml/neural_net/mps_layer_conv_padding.h>
+
+@implementation ConvolutionPadding
+
++ (BOOL) supportsSecureCoding {
+  return true;
+}
+
+- (id) initWithParams:(int)paddingWidth
+        paddingHeight:(int)paddingHeight
+          strideWidth:(int)strideWidth
+         strideHeight:(int)strideHeight {
+  @autoreleasepool {
+    self = [self init];
+    mPaddingWidth = paddingWidth;
+    mPaddingHeight = paddingHeight;
+    mStrideWidth = strideWidth;
+    mStrideHeight = strideHeight;
+    return self;
+  }
+}
+
+- (nullable instancetype)initWithCoder:(NSCoder *)coder {
+  @autoreleasepool {
+    self = [self init];
+    return self;
+  }
+}
+
+- (MPSNNPaddingMethod) paddingMethod {
+  return MPSNNPaddingMethodCustom;
+}
+
+- (MPSImageDescriptor *)destinationImageDescriptorForSourceImages:(NSArray<MPSImage *> *)sourceImages
+                                                     sourceStates:(NSArray<MPSState *> *)sourceStates
+                                                        forKernel:(MPSKernel *)kernel
+                                              suggestedDescriptor:(MPSImageDescriptor *)inDescriptor {
+  MPSCNNConvolution *layer = (MPSCNNConvolution *) kernel;
```

Comment:
If we assume that the kernel is a `MPSCNNConvolution`, can we just use that type in the function signature?

Reply:
If I that's the specific function provided by MPS for me to override don't I have to follow their function signature to the T?